pkg/api/ldap.go

```diff
@@ -215,11 +232,20 @@ func (lc *LDAPClient) Authenticate(username, password string) (bool, map[string]
 	}
 
 	// Bind as the user to verify their password
-	err = lc.Conn.Bind(userDN, password)
-	if err != nil {
-		lc.Log.Error().Err(err).Str("bindDN", userDN).Msg("user bind failed")
+	if password != "" {
+		err = lc.Conn.Bind(userDN, password)
+		if err != nil {
+			lc.Log.Error().Err(err).Str("bindDN", userDN).Msg("user bind failed")
 
-		return false, user, userGroups, err
+			return false, user, userGroups, err
+		}
+	} else {
```

Comment:
What use case are we supporting here? All users we are authenticating should have valid passwords?